tools.py

```diff
@@ -245,6 +261,7 @@ def parseEpisode(title):
       episodenumber = episodeMatch(title)
       showtitle = stripSxxExx(title)
     return [showtitle, episodetitle, seasonnumber, episodenumber, language]
+  
 def compare_and_update(dcmp):
```

Comment:
I had updated the logic here in my script repo to also clean up old or changed files from the provider
`def compare_and_update(dcmp):
    for name in dcmp.diff_files:
        print("STREAM CHANGE -  %s - UPDATING" % (name))
        if os.path.isdir(dcmp.left+"/"+name):
            shutil.copytree(dcmp.left+"/"+name, dcmp.right+"/"+name, dirs_exist_ok=True)
        elif os.path.isfile(dcmp.left+"/"+name):
            shutil.copy2(dcmp.left+"/"+name, dcmp.right+"/"+name)
    for name in dcmp.left_only:
        if os.path.isdir(dcmp.left+"/"+name):
            print("NEW STREAM DIRECTORY - %s - CREATING" % (name))
            shutil.copytree(dcmp.left+"/"+name, dcmp.right+"/"+name, dirs_exist_ok=True)
        elif os.path.isfile(dcmp.left+"/"+name):
            print("NEW STREAM FILE - %s - CREATING" % (name))
            shutil.copy2(dcmp.left+"/"+name, dcmp.right+"/"+name)
    for name in dcmp.right_only:
        if os.path.isdir(dcmp.right+"/"+name):
          print("directory NO LONGER EXISTS - %s - DELETING" % (name))
          shutil.rmtree(dcmp.right+"/"+name)
        if os.path.isfile(dcmp.right+"/"+name):
          print("file NO LONGER EXISTS - %s - DELETING" % (name))
          os.remove(dcmp.right+"/"+name)
    for sub_dcmp in dcmp.subdirs.values():
        compare_and_update(sub_dcmp)
`

Reply:
It has to be merged then, I did a merge before I created the PR so your last changes where in my repo also. I will look into into it, they probably crossed.